tests/components/lock/test_init.py

```diff
@@ -95,6 +107,37 @@ async def test_lock_states(hass: HomeAssistant) -> None:
     assert not lock.is_locked
 
 
+async def test_lock_reg_entry_update(
```

Comment:
Add a tests which tests the behavior when the lock already has the setting in the entity registry options.